patches/server/0728-Add-more-Campfire-API.patch

```diff
@@ -0,0 +1,105 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: LemonCaramel <admin@caramel.moe>
+Date: Fri, 16 Jul 2021 00:39:03 +0900
+Subject: [PATCH] Add more Campfire API
+
+
+diff --git a/src/main/java/net/minecraft/world/level/block/entity/CampfireBlockEntity.java b/src/main/java/net/minecraft/world/level/block/entity/CampfireBlockEntity.java
+index 9f00de75b6f206c11ce7d1c59ba98fafe02fa9ab..9616d6e824f6d8d03d1a0bf099a652afbc490758 100644
+--- a/src/main/java/net/minecraft/world/level/block/entity/CampfireBlockEntity.java
++++ b/src/main/java/net/minecraft/world/level/block/entity/CampfireBlockEntity.java
+@@ -33,12 +33,14 @@ public class CampfireBlockEntity extends BlockEntity implements Clearable {
+     private final NonNullList<ItemStack> items;
+     public final int[] cookingProgress;
+     public final int[] cookingTime;
++    public final byte[] stopCooking; // Paper
+ 
+     public CampfireBlockEntity(BlockPos pos, BlockState state) {
+         super(BlockEntityType.CAMPFIRE, pos, state);
+         this.items = NonNullList.withSize(4, ItemStack.EMPTY);
+         this.cookingProgress = new int[4];
+         this.cookingTime = new int[4];
++        this.stopCooking = new byte[4]; // Paper
+     }
+ 
+     public static void cookTick(Level world, BlockPos pos, BlockState state, CampfireBlockEntity campfire) {
+@@ -49,7 +51,9 @@ public class CampfireBlockEntity extends BlockEntity implements Clearable {
+ 
+             if (!itemstack.isEmpty()) {
+                 flag = true;
++                if (campfire.stopCooking[i] == 0) { // Paper
+                 int j = campfire.cookingProgress[i]++;
++                } // Paper
+ 
+                 if (campfire.cookingProgress[i] >= campfire.cookingTime[i]) {
+                     SimpleContainer inventorysubcontainer = new SimpleContainer(new ItemStack[]{itemstack});
+@@ -156,6 +160,12 @@ public class CampfireBlockEntity extends BlockEntity implements Clearable {
+             System.arraycopy(aint, 0, this.cookingTime, 0, Math.min(this.cookingTime.length, aint.length));
+         }
+ 
++        // Paper start
++        if (nbt.contains("Paper.StopCooking", 7)) { // 7: Byte Array
```

Comment:
Instead of using the magic value `7`, in `CraftMagicNumbers.NBT`, there are a bunch of static values you should use here.